test/Microsoft.AspNet.TestHost.Tests/ClientHandlerTests.cs

```diff
@@ -269,5 +275,61 @@ public Task ProcessRequestAsync(Context context)
                 return _application(context.HttpContext);
             }
         }
+
+
+        [ConditionalFact]
+        [FrameworkSkipCondition(RuntimeFrameworks.Mono, SkipReason = "Hangs randomly (issue #507)")]
+        public async Task ClientHandlerCreateContextWithDefaultRequestParameters()
+        {
+            var logger = new VerifierLogger();
+            var builder = new WebApplicationBuilder()
+                            .ConfigureServices(services =>
+                            {
+                                services.AddSingleton<ILogger<WebApplication>>(logger);
+                            })
+                            .Configure(app =>
+                            {
+                                app.Run(context =>
+                                {
+                                    return Task.FromResult(0);
+                                });
+                            });
+            var server = new TestServer(builder);
+
+            var result = await server.CreateClient().GetStringAsync("/");
+            Assert.Contains("Request starting HTTP/1.1 GET http://", logger.Message);
```

Comment:
This is actually pretty misleading. We are only using the logger to access all the fields of the request to make sure they are set when creating the context. It's meaningless as we set them to different values after the creation.